x/crosschain/keeper/msg_server_refund_aborted_tx.go

```diff
@@ -0,0 +1,93 @@
+package keeper
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/zeta-chain/zetacore/common"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+	"golang.org/x/net/context"
+)
+
+func (k msgServer) RefundAbortedCCTX(goCtx context.Context, msg *types.MsgRefundAbortedCCTX) (*types.MsgRefundAbortedCCTXResponse, error) {
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check if authorized
+	if msg.Creator != k.zetaObserverKeeper.GetParams(ctx).GetAdminPolicyAccount(observertypes.Policy_Type_group2) {
+		return nil, observertypes.ErrNotAuthorized
+	}
+
+	// check if the cctx exists
+	cctx, found := k.GetCrossChainTx(ctx, msg.CctxIndex)
+	if !found {
+		return nil, types.ErrCannotFindCctx
+	}
+
+	// check if the cctx is aborted
+	if cctx.CctxStatus.Status != types.CctxStatus_Aborted {
+		return nil, errorsmod.Wrap(types.ErrInvalidStatus, "CCTX is not aborted")
+	}
+	// check if the cctx is not refunded
+	if cctx.CctxStatus.IsAbortRefunded {
+		return nil, errorsmod.Wrap(types.ErrUnableProcessRefund, "CCTX is already refunded")
+	}
+
+	// Check if aborted amount is available to maintain zeta accounting
+	// NOTE: Need to verify if this check works / is required in athens 3
```

Comment:
Any issue we can create for this? Why would there be issues?

Reply:
cd830fb1c6060e0de855bd388cb54b4fd58a7e36